### PR TITLE
Update "handlebars" to v4.0.6

### PR DESCRIPTION
**Pull request by Turbo87**
related to #224 

/cc @mmun @rwjblue

### CODE REVIEW

**Comment by rwjblue** [comment]
@Turbo87 - LGTM, can you prepare the corresponding Ember PR also and confirm all works well?

**Comment by rwjblue** [comment]
LGTM, I'd like @mmun to review / confirm also.

**Comment by rwjblue** [comment]
I think we should address the things listed in https://github.com/tildeio/glimmer/issues/224 before landing this.  

From @mmun in that issue:

> Partials (specifically the {{> foo}} syntax and its variations and decorators. They weren't designed with glimmer in mind and it's not clear at this time how they fit into our component story.

File: packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts [hunk 1]
```diff
@@ -136,8 +135,15 @@ export default {
   },
 
   PartialStatement: function(partial) {
-    appendChild(this.currentElement(), partial);
-    return partial;
+    let { name, loc } = partial;
```
Comment (rafde) [inline packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts]: Will decorators work?

Comment (rwjblue) [inline packages/@glimmer/syntax/lib/parser/handlebars-node-visitors.ts]: No, we will throw for PartialStatement, Decorator, DecoratorBlock. Possibly more, but at least these.

File: packages/@glimmer/syntax/tests/parser-node-test.ts [hunk 2]
```diff
@@ -470,3 +470,15 @@ test("allow {{undefined}} to be passed as a param", function() {
     b.mustache(b.path('foo'), [b.undefined()])
   ]));
 });
+
+test("Handlebars partial should error", function(assert) {
+  assert.throws(() => {
+    parse("{{> foo}}");
+  }, Error(`Handlebars partials are not supported: "{{> foo" on line "1:0"`));
```
Comment (Turbo87) [inline packages/@glimmer/syntax/tests/parser-node-test.ts]: @locks IMHO `on line "1:0"` looks a bit weird.

- if we say `on line` we should just print the line but not the column
- if we said `at "1:0"` people might not understand that we are talking about a code location

maybe just append `(at 1:0)` at the end?

Comment (rwjblue) [inline packages/@glimmer/syntax/tests/parser-node-test.ts]: Ya, we should say `on line X column Z` or `at L1:C2` (we do the latter in a number of ember's internal errors/warnings).

Comment (locks) [inline packages/@glimmer/syntax/tests/parser-node-test.ts]: I was copying https://github.com/tildeio/glimmer/pull/397/files#diff-96d3a15c8247549ce9be44ee089dc869L132. I'll do `at L1:C2` if Ember has that.

File: packages/@glimmer/syntax/tests/parser-node-test.ts [hunk 3]
```diff
@@ -474,11 +474,23 @@ test("allow {{undefined}} to be passed as a param", function() {
...
+test("Handlebars decorator should error", function(assert) {
+  assert.throws(() => {
+    parse("{{* foo}}");
+  }, new Error(`Handlebars decorators are not supported: "{{* foo" at L1:C0`));
+});
+
+test("Handlebars decorator block should error", function(assert) {
+  assert.throws(() => {
+    parse("{{#* foo}}{{/foo}}");
+  }, new Error(`Handlebars decorator blocks are not supported: "{{#* foo}}{{/foo}}" at L1:C0`);
```
Comment (locks) [inline packages/@glimmer/syntax/tests/parser-node-test.ts]: Not 100% happy with the error message, seems a bit confusing to include the closing tag? I'll tweak. Please comment 😁

Comment (rwjblue) [inline packages/@glimmer/syntax/tests/parser-node-test.ts]: Seems fine to me.

Comment (locks) [inline packages/@glimmer/syntax/tests/parser-node-test.ts]: Updated to match partials, let me know if I'm missing anyway, otherwise seems good to go.

**Comment by rwjblue** [comment]
Awesome, thank you @Turbo87 and @locks!

**Pull request merged**